d2target/d2target.go

```diff
@@ -72,6 +73,66 @@ type Diagram struct {
 	Steps     []*Diagram `json:"steps,omitempty"`
 }
 
+// boardPath comes in the form of "x/layers/z/scenarios/a"
+// or in the form of "layers/z/scenarios/a"
+func (d *Diagram) GetBoard(boardPath string) *Diagram {
+	path := strings.Split(boardPath, string(os.PathSeparator))
+	if len(path) == 0 || len(boardPath) == 0 {
+		return d
+	}
+
+	head := path[0]
+
+	if head == "index" {
+		return d
+	}
+
+	switch head {
+	case "layers", "scenarios", "steps":
+		if len(path) < 2 {
+			return nil
+		}
+	}
+
+	switch head {
+	case "layers":
+		for _, b := range d.Layers {
+			if b.Name == path[1] {
+				return b.GetBoard(strings.Join(path[2:], string(os.PathSeparator)))
```

Comment:
Add and reimplement in terms of an unexported `getBoard` that takes the path as an array to avoid this.